src/php/ext/grpc/channel.c

```diff
@@ -223,9 +346,50 @@ PHP_METHOD(Channel, watchConnectivityState) {
 PHP_METHOD(Channel, close) {
   wrapped_grpc_channel *channel = Z_WRAPPED_GRPC_CHANNEL_P(getThis());
   if (channel->wrapped != NULL) {
-    grpc_channel_destroy(channel->wrapped);
+    // this fixes a race when two shared Channel objects both call close()
+    // at about the same time
+    int state = grpc_channel_check_connectivity_state(channel->wrapped, 0);
```

Comment:
This seems like a really awkward way to avoid a race condition.

It seems like the same thing could be accomplished using a mutex, along with a small shared wrapper object around the ```grpc_channel*```.